client/image_list_test.go

```diff
@@ -163,62 +165,37 @@ func TestImageListApiBefore125(t *testing.T) {
 // Checks if shared-size query parameter is set/not being set correctly
 // for /images/json.
 func TestImageListWithSharedSize(t *testing.T) {
-	// performs ImageList call and returns the sent url query
-	getImageListAndGetQuery := func(version string, options types.ImageListOptions) (query url.Values, err error) {
-		client := &Client{
-			client: newMockClient(func(req *http.Request) (*http.Response, error) {
-				query = req.URL.Query()
-
-				return &http.Response{
-					StatusCode: http.StatusOK,
-					Body:       io.NopCloser(strings.NewReader("[]")),
-				}, nil
-			}),
-			version: version,
-		}
-
-		_, err = client.ImageList(context.Background(), options)
-		return query, err
-	}
-
+	t.Parallel()
 	const sharedSize = "shared-size"
-
-	// Check if the shared-size query parameter is unset
-	for _, tC := range []struct {
-		name    string
-		version string
-		options types.ImageListOptions
+	for _, tc := range []struct {
+		name       string
+		version    string
+		options    types.ImageListOptions
+		sharedSize string // value the shared-size query param should be set to, or empty if it should not be set.
 	}{
 		{name: "after 1.42, if not requested", version: "1.42", options: types.ImageListOptions{SharedSize: false}},
+		{name: "after 1.42, if requested", version: "1.42", options: types.ImageListOptions{SharedSize: true}, sharedSize: "1"},
 		{name: "before 1.42, even if requested", version: "1.41", options: types.ImageListOptions{SharedSize: true}},
 	} {
-		t.Run("unset "+tC.name, func(t *testing.T) {
-			query, err := getImageListAndGetQuery(tC.version, tC.options)
-			if err != nil {
-				t.Error(err)
-			}
-
-			if query.Has(sharedSize) {
-				t.Errorf("expected %q to be unset", sharedSize)
+		tc := tc
+		t.Run("unset "+tc.name, func(t *testing.T) {
```

Comment:
Now it's not always "unset", so I'll move this to the "name" field, depending on the case.
Just a note to self, will fix it when squashing :)

Reply:
Good catch; updated 👍